public/pages/Overview/containers/Overview/Overview.tsx

```diff
@@ -0,0 +1,199 @@
+import { compile } from 'vega-lite';
+import { ROUTES } from '../../../../utils/constants';
+import { FindingItem, OverviewProps, OverviewState } from '../../types/interfaces';
+import { dummyWidgetItems, getVisualizationSpec } from '../../utils/dummyData';
+import { groupByOptions, widgetHeaderData } from '../../utils/constants';
+
+export default class Overview extends Component<OverviewProps, OverviewState> {
+  constructor(props: OverviewProps) {
+    super(props);
+    this.state = {
+      groupBy: 'all_findings',
+    };
+  }
+
+  createWidgetHeader(title: string, widgetActions: JSX.Element | null) {
+    return (
+      <EuiFlexItem>
+        <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
+          <EuiFlexItem>
+            <EuiText size="m">
+              <h4>{title}</h4>
+            </EuiText>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>{widgetActions}</EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiFlexItem>
+    );
+  }
+
+  createActionBtn(name: string, onClick: () => void) {
+    return (
+      <EuiButton size="s" onClick={onClick}>
+        {name}
+      </EuiButton>
+    );
+  }
+
+  createSelectComponent(
+    options: EuiSelectOption[],
+    value: string,
+    onChange: React.ChangeEventHandler<HTMLSelectElement>
+  ) {
+    return (
+      <EuiFlexGroup justifyContent="flexEnd" alignItems="center">
+        <EuiFlexItem grow={false}>
+          <h5>Group by</h5>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiSelect
+            id="overview-vis-options"
+            options={options}
+            value={value}
+            onChange={onChange}
+          />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  }
+
+  generateVisualizationSpec() {
+    return getVisualizationSpec();
+  }
+
+  renderVis() {
+    let view;
+    const spec = this.generateVisualizationSpec();
+
+    try {
+      renderVegaSpec(compile({ ...spec, width: 'container', height: 400 }).spec).catch((err) =>
+        console.error(err)
+      );
+    } catch (error) {
+      console.log(error);
+    }
+
+    function renderVegaSpec(spec: {}) {
+      view = new View(parse(spec), {
+        renderer: 'canvas', // renderer (canvas or svg)
+        container: '#view', // parent DOM container
+        hover: true, // enable hover processing
+      });
+      return view.runAsync();
+    }
+  }
+
+  componentDidMount(): void {
+    this.renderVis();
+  }
+
+  render() {
+    const columns: EuiBasicTableColumn<FindingItem>[] = [
+      {
+        field: 'time',
+        name: 'Time',
+        sortable: true,
+        align: 'left',
+      },
+      {
+        field: 'findingName',
+        name: 'Finding Name',
+        sortable: false,
+        align: 'left',
+      },
+      {
+        field: 'detector',
+        name: 'Detector',
+        sortable: true,
+        align: 'left',
+      },
+    ];
+
+    return (
+      <ContentPanel title={'Overview'}>
```

Comment:
The code here looks good; this is more of a style comment. The mocks suggest the graphs and tables shouldn't be wrapped in one `ContentPanel`. Instead, each graph should have its own content panel. The content panel `title` attribute should then replace the need for the `createWidgetHeader` method.

This isn't a blocking issue for this prototype though.

Reply:
Makes sense, let me change that in a follow up PR when connecting UI to the backend as part of [this feature](https://github.com/opensearch-project/security-analytics-dashboards-plugin/issues/12)